frontend/src/scenes/surveys/surveyLogic.tsx

```diff
@@ -1256,17 +1263,28 @@ export const surveyLogic = kea<surveyLogicType>([
                         return 'No data available'
                     }
 
-                    const data: number[] = questionResults.data
-                    if (data.length === 11) {
-                        const promoters = data.slice(9, 11).reduce((a, b) => a + b, 0)
-                        const passives = data.slice(7, 9).reduce((a, b) => a + b, 0)
-                        const detractors = data.slice(0, 7).reduce((a, b) => a + b, 0)
-                        const npsScore = ((promoters - detractors) / (promoters + passives + detractors)) * 100
-                        return npsScore.toFixed(1)
+                    const npsBreakdown = calculateNpsBreakdown(questionResults)
+                    if (!npsBreakdown) {
+                        return null
                     }
+
+                    return calculateNpsScore(npsBreakdown).toFixed(1)
                 }
             },
         ],
+        npsBreakdown: [
+            (s) => [s.surveyRatingResults],
+            (surveyRatingResults) => {
+                const surveyRatingKeys = Object.keys(surveyRatingResults ?? {})
+                if (surveyRatingKeys.length === 0) {
+                    return null
+                }
+                const questionIdx = surveyRatingKeys[0]
+                const questionResults = surveyRatingResults[questionIdx]
```

Comment:
could it be that`surveyRatingResults[questionIdx]` is out of range? should we check the length based on the index?

Reply:
`surveyRatingResults` is not an array, it's an object that uses the question index as the key:

```ts
export interface SurveyRatingResults {
    [key: number]: {
        data: number[]
        total: number
    }
}
```

so checking for out of range here doesn't make sense

Reply:
true, i'll make this check on the survey logic before calling `calculateNpsBreakdown`